### PR TITLE
fix connect timeout while getting IAM creds in Docker

### PR DESCRIPTION
**Pull request by aiudirog**
### Description of Change

While trying to use S3FS in a Docker container on my EC2, I found that `AioIMDSFetcher._fetch_metadata_token()` was throwing a `ConnectTimeoutError` trying to connect to the metadata service with `PUT http://169.254.169.254/latest/api/token/`. This came from an `aiohttp.ServerTimeoutError` while trying to submit the PUT.

 When I compared the logs to plain botocore, I found that it instead encountered a `ReadTimeoutError`, which it ignored and moved on to trying `GET http://169.254.169.254/latest/meta-data/iam/security-credentials/`, which succeeded. This error was redirected from a `urllib3.ReadTimeoutError` instead of a `ConnectTimeoutError`.

 I believe this particular difference comes from aiohttp and urllib3 handling the server connection issue differently. By handling the `ConnectTimeoutError`, aiobotocore is able to successfully get the credentials from the metadata service while in Docker.

 Note that this issue does not occur outside of Docker, even with an identical virtual env.

### Assumptions

- A `ConnectTimeoutError` from the metadata service shouldn't bubble up and end attempts to resolve credentials

### Checklist for All Submissions
* [x] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)
* [x] If this is resolving an issue (needed so future developers can determine if change is still necessary and under what conditions) (can be provided via link to issue with these details):
  * [ ] Detailed description of issue
  * [ ] Alternative methods considered (if any)
  * [ ] How issue is being resolved
  * [ ] How issue can be reproduced 
* [ ] If this is providing a new feature  (can be provided via link to issue with these details):
  * [ ] Detailed description of new feature
  * [ ] Why needed
  * [ ] Alternatives methods considered (if any)

### Checklist when updating botocore and/or aiohttp versions

* [x] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [x] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [x] I have ensured that the awscli/boto3 versions match the updated botocore version

### CODE REVIEW

File: aiobotocore/utils.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ async def _fetch_metadata_token(self):
                         return None
                     elif response.status_code in (400,):
                         raise BadIMDSRequestError(request)
-                except ReadTimeoutError:
+                except (ReadTimeoutError, ConnectTimeoutError):
```
Comment (thehesiod) [inline aiobotocore/utils.py:95]: this is probably actually an issue with our exception mapping here: https://github.com/aio-libs/aiobotocore/blob/master/aiobotocore/httpsession.py#L197 mind investigating?

Comment (thehesiod) [reply aiobotocore/utils.py:95]: ya see: https://github.com/boto/botocore/blob/develop/botocore/utils.py#L461 I'm pretty sure we're just not mapping the exceptions correctly, we should probably port that relevant test over as well

Comment (aiudirog) [reply aiobotocore/utils.py:95]: Now that you mention it, I believe you're right: https://docs.aiohttp.org/en/stable/client_reference.html?highlight=servertimeouterror#aiohttp.ServerTimeoutError

The aiohttp docs say that `ServerTimeoutError` is also for read timeouts! I'll update my PR to reflect that.

---------------

Also, wow, that was the fastest response I've ever gotten to a PR! Thanks:)

Comment (aiudirog) [reply aiobotocore/utils.py]: Updated!

File: aiobotocore/httpsession.py [hunk 2]
```diff
@@ -205,12 +205,10 @@ async def send(self, request):
                 request=request,
                 endpoint_url=request.url
             )
-        except ServerTimeoutError as e:
-            raise ConnectTimeoutError(endpoint_url=request.url, error=e)
```
Comment (thehesiod) [inline aiobotocore/httpsession.py:209]: we still do need to support ConnectTimeoutError: https://github.com/boto/botocore/blob/develop/botocore/httpsession.py#L478. so probably need something a little more refined

File: aiobotocore/utils.py (hunk 1 above)
Comment (thehesiod) [reply aiobotocore/utils.py]: Welcoming a little distraction from this issue I'm working on :) added reply

File: aiobotocore/httpsession.py (hunk 2 above)
Comment (aiudirog) [reply aiobotocore/httpsession.py:209]: Digging through the source code, it looks like we can differentiate by looking at the message:
- Connection timeout - https://github.com/aio-libs/aiohttp/blob/8e92c4c7c305f914c17f41539da20822ea7baddf/aiohttp/client.py#L505
- Read timeout - https://github.com/aio-libs/aiohttp/blob/8e92c4c7c305f914c17f41539da20822ea7baddf/aiohttp/client_proto.py#L197

How about:

```python
if str(e).lower().startswith('connect'):
    raise ConnectTimeoutError(...)
else:
    raise ReadTimeoutError(...)
```

**Comment by aiudirog** [comment]
@thehesiod Are there any other changes you would like me to make?

**Comment by thehesiod** [comment]
my apologies for delay, just been crazy here

File: CHANGES.rst [hunk 3]
```diff
@@ -1,5 +1,9 @@
 Changes
 -------
+2.3.4 (2022-05-20)
```
Comment (thehesiod) [inline CHANGES.rst:3]: forgot to bump version in `__init__.py`

Comment (aiudirog) [reply CHANGES.rst:3]: Updated!

**Pull request merged**